src/DataTransferObject.php

```diff
@@ -7,92 +7,206 @@
 use ReflectionClass;
 use ReflectionProperty;
 
-abstract class DataTransferObject
+/**
+ * Class DataTransferObject.
+ */
+abstract class DataTransferObject implements DtoContract
 {
     /** @var array */
     protected $exceptKeys = [];
 
     /** @var array */
     protected $onlyKeys = [];
 
+    /** @var Property[] | array */
+    protected $properties = [];
+
+    /** @var bool */
+    protected $immutable;
+
+    public static function mutable(array $parameters): self
+    {
+        return new static($parameters, false);
+    }
+
+    public static function immutable(array $parameters): self
+    {
+        return new static($parameters, true);
+    }
+
+    public function __construct(array $parameters, bool $immutable = true)
```

Comment:
We shouldn't handle immutability via the constructor. Because this package is meant to be extended by concrete DTO classes, we want to constructor to be as clean as possible.

Let's add a fluent setter and use it like so:

```php
public static function immutable(array $parameters): self
{
    return (new static($parameters, true))->setImmutable(true);
}
```